pkg/envoy/sds/response.go

```diff
@@ -145,58 +144,66 @@ func (s *sdsImpl) getRootCert(cert certificate.Certificater, sdscert envoy.SDSCe
 		return secret, nil
 	}
 
-	// Program SAN matching based on SMI TrafficTarget policies
-	switch sdscert.CertType {
-	case envoy.RootCertTypeForMTLSOutbound:
+	svcIdentities, err := getServiceIdentitiesFromCert(sdsCert, s.serviceIdentity, s.meshCatalog)
+	if err != nil {
+		return nil, err
+	}
+
+	secret.GetValidationContext().MatchSubjectAltNames = getSubjectAltNamesFromSvcIdentities(svcIdentities)
+	return secret, nil
+}
+
+func getServiceIdentitiesFromCert(sdsCert envoy.SDSCert, svcIdentity identity.ServiceIdentity, mc catalog.MeshCataloger) ([]identity.ServiceIdentity, error) {
+	// Configure SAN matching based on SMI TrafficTarget policies
+	if sdsCert.CertType == envoy.RootCertTypeForMTLSOutbound {
 		// For the outbound certificate validation context, the SANs needs to match the list of service identities
-		// corresponding to the upstream service. This means, if the sdscert.Name points to service 'X',
+		// corresponding to the upstream service. This means, if the sdsCert.Name points to service 'X',
 		// the SANs for this certificate should correspond to the service identities of 'X'.
-		meshSvc, err := service.UnmarshalMeshService(sdscert.Name)
+		meshSvc, err := service.UnmarshalMeshService(sdsCert.Name)
 		if err != nil {
-			log.Error().Err(err).Msgf("Error unmarshalling upstream service for outbound cert %s", sdscert)
+			log.Error().Err(err).Msgf("Error unmarshalling upstream service for outbound cert %s", sdsCert.Name)
 			return nil, err
 		}
-		svcIdentities, err := s.meshCatalog.ListServiceIdentitiesForService(*meshSvc)
+		svcIdentities, err := mc.ListServiceIdentitiesForService(*meshSvc)
 		if err != nil {
 			log.Error().Err(err).Msgf("Error listing service accounts for service %s", meshSvc)
 			return nil, err
 		}
-		secret.GetValidationContext().MatchSubjectAltNames = getSubjectAltNamesFromSvcAccount(svcIdentities)
+		return svcIdentities, nil
+	}
 
-	case envoy.RootCertTypeForMTLSInbound:
+	if sdsCert.CertType == envoy.RootCertTypeForMTLSInbound {
```

Comment:
Instead of a bunch of `if` clauses with returns, a `switch-case` cluase reads better to me. Since the purpose of this PR is to carve out a new function `getServiceIdentitiesFromCert()`, I would avoid refactoring other parts of the code here since it only makes it harder to review.

Reply:
Sure, if you prefer `switch`, I'll keep `switch`.